### PR TITLE
Archive

### PR DESCRIPTION
**Pull request by ahohnsen**

### CODE REVIEW

File: src/pages/ArchivePage.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  onEditDiveWish,
+  onDeleteDiveWish,
+}) {
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    if (archivedWishes.length === 0) {
+      navigate('/');
+    }
+  }, [archivedWishes, navigate]);
```
Comment (mwarnck) [inline src/pages/ArchivePage.js:22]: What is this code for? To directly get to the start page when deleted the last archived item?

File: src/pages/ArchivePage.spec.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      {
+        id: '2',
+        destination: 'Galapagos Island',
+        notes: 'My dream destination but first I have to win the lottery.',
+        isArchived: true,
+      },
+    ];
+
+    render(
+      <MemoryRouter>
```
Comment (mwarnck) [inline src/pages/ArchivePage.spec.js:23]: Do you need the MemoryRouter for testing the single component ArchivePage and to render the content?

**Review by PReichetanz — COMMENTED** [review]
Nice work, but quite a big PR! I only have minor issues to think about 🚀

**Review by mwarnck — COMMENTED** [review]
I really like your stories with all the different versions of the component ;) I left only two little comments / questions for your code :)

File: src/pages/ArchivePage.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (macfrei) [reply src/pages/ArchivePage.js:22]: Have a look at this article: https://www.robinwieruch.de/react-router-redirect/ 
I think the first two solutions would be a bit cleaner than the use of `useEffect`.

[the remark this answers is not in the document]
Comment (ahohnsen) [reply src/pages/ArchivePage.js:22]: @ Paul unfortunately, this was a suggestion from ESLint... I added it and didn't find any issue, all was working fine. Alternative would be to set this line manually on ignore for ESLint.

@ Lene thx, I will have a look

Comment (ahohnsen) [reply src/pages/ArchivePage.js:22]: Exactly! As I only want the ArchivePage to be visible if there is already an archived item. Without that code, the user would have to go back manually once he/she takes off the last item of the archive page

File: src/pages/ArchivePage.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        </Grid>
+      </Content>
+    </>
+  );
+}
+
+const Grid = styled.div`
+  display: grid;
+  gap: 15px;
+`;
```
[the remark this answers is not in the document]
Comment (ahohnsen) [reply src/pages/ArchivePage.js:56]: I tried that on another page beforehand but always ran into issues with the overflow and margins. Couldn't fix that so far.

File: src/pages/ArchivePage.spec.js (hunk 2 above)
Comment (ahohnsen) [reply src/pages/ArchivePage.spec.js:23]: Yes, as I have the useNavigate hook inside the ArchivePage. Without the MemoryRouter I will get an error while testing

File: src/App.js [hunk 4]
```diff
@@ -66,12 +66,16 @@ export default function App() {
...
-            />
+            archivedWishes.length === 0 ? (
+              <Navigate replace to="/" />
+            ) : (
+              <ArchivePage
+                archivedWishes={archivedWishes}
+                onToggleCheckmark={toggleCheckmark}
+                onEditDiveWish={handleEditRedirect}
+                onDeleteDiveWish={handleDeleteWish}
+              />
```
Comment (macfrei) [inline src/App.js:77]: Nice :)

**Pull request merged**